srcpkgs/ffmpeg7/template

```diff
@@ -0,0 +1,202 @@
+				--enable-libmp3lame --enable-libvorbis --enable-libxvid \
+				--enable-libx264 --enable-libvpx --enable-libtheora \
+				--enable-shared --enable-static --enable-libxcb \
+				--enable-libpulse --enable-libfreetype --enable-libmodplug \
+				--enable-libspeex --enable-libcelt --enable-libass \
+				--enable-libopus --enable-librtmp --enable-libjack \
+				--disable-libopencore_amrnb --disable-libopencore_amrwb \
+				--disable-libopenjpeg --enable-libbluray \
+				--enable-postproc --enable-opencl --enable-libvmaf ${_args} \
+				--enable-libx265 --enable-libv4l2 --enable-libaom \
+				--enable-libbs2b --enable-libvidstab --enable-libdav1d \
+				--enable-libsrt --enable-librist --enable-libwebp \
+				--enable-vulkan --enable-libdrm --enable-libsvtav1 \
+				--enable-libfreetype --enable-libharfbuzz --enable-libfontconfig \
+				$(vopt_if fdk_aac '--enable-nonfree --enable-libfdk-aac') \
+				$(vopt_enable vaapi) $(vopt_enable vdpau) \
+				$(vopt_enable zimg libzimg) \
+				$(vopt_enable sofa libmysofa) \
+				$(vopt_enable onevpl libvpl) \
+				$(vopt_enable nvcodec nvenc) \
+				$(vopt_enable nvcodec nvdec)
+
+}
+
+do_build() {
+		make ${makejobs}
+		make doc/ff{mpeg,play}.1
+}
+
+do_install() {
+		make DESTDIR=${DESTDIR} install install-man
+}
+
+libavcodec7_package() {
+		short_desc="FFmpeg codec library"
+		pkg_install() {
+				vmove "usr/lib/libavcodec.so.*"
+		}
+}
+
+libavdevice7_package() {
+		short_desc="FFmpeg device handling library"
+		pkg_install() {
+				vmove "usr/lib/libavdevice.so.*"
+		}
+}
+
+libavresample7_package() {
+		short_desc="Package deprecated, removal needed"
+		build_style=meta
+}
+
+libavformat7_package() {
+		short_desc="FFmpeg file format library"
+		pkg_install() {
+				vmove "usr/lib/libavformat.so.*"
+		}
+}
+
+libavutil7_package() {
+		short_desc="FFmpeg utility library"
+		pkg_install() {
+				vmove "usr/lib/libavutil.so.*"
+		}
+}
+
+libavfilter7_package() {
+		short_desc="FFmpeg audio/video filter library"
+		pkg_install() {
+				vmove "usr/lib/libavfilter.so.*"
+		}
+}
+
+libpostproc7_package() {
+		short_desc="FFmpeg video postprocessing library"
+		pkg_install() {
+				vmove "usr/lib/libpostproc.so.*"
+		}
+}
+
+libswscale7_package() {
+		short_desc="FFmpeg video scaling library"
+		pkg_install() {
+				vmove "usr/lib/libswscale.so.*"
+		}
+}
+
+libswresample7_package() {
+		short_desc="FFmpeg video resampling library"
+		pkg_install() {
+				vmove "usr/lib/libswresample.so.*"
+		}
+}
+
+ffmpeg7-devel_package() {
+		depends="
+				libavcodec7>=${version}_${revision}
+				libavdevice7>=${version}_${revision}
+				libavformat7>=${version}_${revision}
+				libavutil7>=${version}_${revision}
+				libavfilter7>=${version}_${revision}
+				libpostproc7>=${version}_${revision}
+				libswscale7>=${version}_${revision}
+				libswresample7>=${version}_${revision}"
+		short_desc+=" - development files"
+		conflicts="ffmpeg-devel"
+		replaces="ffmpeg-devel>=0"
```

Comment:
needs to conflict/replace with other ffmpeg*-devel too

Reply:
Is this necessary for the other subpkgs as well? The build of the ffplay7 package failed due to the conflict field.

```diff
...
ffplay7_package() {
		short_desc="Simple video player using FFmpeg and SDL2"
+		conflicts="ffplay*"
+		replaces="ffplay*>0"
		pkg_install() {
				vmove usr/bin/ffplay
				vmove "usr/share/man/man1/ffplay*"
		}
}
```

Reply:
think ffplay is fine but ffmpeg6-devel exist

Reply:
may need to disable ffplay in 6...or replace 6 entirely idk, depends how many deps build with 7 fine?

Reply:
> may need to disable ffplay in 6...or replace 6 entirely
Why are you considering disabling `ffplay` in 6? Is there a specific reason behind it?

Reply:
7 is newer so the binary ffmpeg/ffplay may as well be 7 on the system. only a few thing use the binaries directly instead of the libraries. ill post my branch iaf.

basically:
- add depends to ffplay7 here
- `ffmpeg` and `ffmpeg6` need the conflict for devel packages and a revbumped
- `ffplay6` becomes meta pkg like `ffmpeg`'s `ffplay` is